core/kernel/thread.c

```diff
@@ -93,6 +94,25 @@ void thread_init_canaries(void)
 #endif/*CFG_WITH_STACK_CANARIES*/
 }
 
+#if defined(CFG_WITH_STACK_CANARIES)
+void thread_update_canaries(void)
+{
+	uint32_t canary[2] = { };
+	uint32_t exceptions = 0;
+
+	plat_get_random_stack_canaries(canary, ARRAY_SIZE(canary),
+				       sizeof(canary[0]));
+
+	exceptions = thread_mask_exceptions(THREAD_EXCP_ALL);
```

Comment:
Should canaries be checed before there are updated?

Reply:
Are you suggesting that we should check the canaries are assigned random values correctly by plat_get_random_stack_canaries, rather than the initial value of '0'?

Reply:
No, I mean verifiying that canaries used (load in stacks) before this function execute, are still alive before we scratch them with new canary values.